src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Formatting/ISyntaxFormattingService.cs

```diff
@@ -22,27 +22,41 @@ internal interface ISyntaxFormattingService
         : ILanguageService
 #endif
     {
+        SyntaxFormattingOptions GetFormattingOptions(AnalyzerConfigOptions options);
         IEnumerable<AbstractFormattingRule> GetDefaultFormattingRules();
         IFormattingResult GetFormattingResult(SyntaxNode node, IEnumerable<TextSpan>? spans, SyntaxFormattingOptions options, IEnumerable<AbstractFormattingRule>? rules, CancellationToken cancellationToken);
     }
 
-    internal readonly record struct SyntaxFormattingOptions(
-        AnalyzerConfigOptions Options)
+    internal abstract class SyntaxFormattingOptions
     {
-        public static readonly SyntaxFormattingOptions Default = Create(DictionaryAnalyzerConfigOptions.Empty);
+        public readonly bool UseTabs;
+        public readonly int TabSize;
+        public readonly int IndentationSize;
+        public readonly string NewLine;
```

Comment:
i'd actually love for these four to be their own struct.

Reply:
Possibly. We can do another round of refactorings/renames once finished with all options.

Reply:
https://github.com/dotnet/roslyn/issues/59090